File: 2024/CountElementsWithStrictlySmallerAndGreaterElements.java
// Time complexity - O (N)
// Space complexity - O (1)
public class CountElementsWithStrictlySmallerAndGreaterElements {

    public int countElements(int[] nums) {
        int result = 0;
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;

        for (int num : nums) {
            min = Math.min(min, num);
            max = Math.max(max, num);
        }

        for (int num : nums) {
            if (num > min && num < max) {
                result++;
            }
        }

        return result;
    }
}

File: python/count _elements_with_strictly_smaller_and_greater_elements.py
from typing import List


# Time complexity - O (N)
# Space complexity - O (1)
class Solution:
    def countElements(self, nums: List[int]) -> int:
        result = 0
        min_val = float('inf')
        max_val = float('-inf')

        for num in nums:
            min_val = min(min_val, num)
            max_val = max(max_val, num)

        for num in nums:
            if min_val < num < max_val:
                result += 1

        return result